packages/knack-avro/src/encodeDecode.js

```diff
@@ -9,6 +9,16 @@ const options = {
 	mapDecoded: undefined
 };
 
+const collectInvalidPaths = (type, val) => {
```

Comment:
very cool!

Reply:
suggested addition: include an env flag to allow devs to shut off better errors in production scenarios.